From 1a26b21b309d263ad00977477ed638c909f28702 Mon Sep 17 00:00:00 2001
From: Boerje Sewing <Boerje.Sewing@who-ing.de>
Date: Mon, 12 Feb 2018 11:42:08 +0100
Subject: [PATCH] dts: src: add kbus io pins.

* configure kbus io pins as gpios, spi and uart
* add src specific kbus spi device to spidev driver
* this is on purpose: at this time no kbus device is desired

Signed-off-by: Boerje Sewing <Boerje.Sewing@who-ing.de>
---
 arch/arm/boot/dts/wago-src.dtsi | 60 +++++++++++++++++++++++++++++------------
 drivers/spi/spidev.c            |  1 +
 2 files changed, 44 insertions(+), 17 deletions(-)

diff --git a/arch/arm/boot/dts/wago-src.dtsi b/arch/arm/boot/dts/wago-src.dtsi
index 23153da..7d594cc 100644
--- a/arch/arm/boot/dts/wago-src.dtsi
+++ b/arch/arm/boot/dts/wago-src.dtsi
@@ -217,6 +217,8 @@
 			0x0d4 ( MUX_MODE7 | PIN_INPUT		) /* V3 LCD_DATA13		7   GPIO0_9->SYSBOOT13*/
 			0x0d8 ( MUX_MODE7 | PIN_INPUT		) /* V4 LCD_DATA14		7   GPIO0_10->SYSBOOT14*/
 			0x0dc ( MUX_MODE7 | PIN_INPUT		) /* T5 LCD_DATA15		7   GPIO0_11->SYSBOOT15*/
+			0x124 ( MUX_MODE7 | PIN_INPUT_PULLUP	) /* K16 GMII1_TXD1		7   GPIO0_21->KBUS-nRDY */
+			0x128 ( MUX_MODE7 | PIN_INPUT_PULLUP 	) /* K17 GMII1_TXD0		7   GPIO0_28->KBUS-CMDSEL */
 			0x178 ( MUX_MODE7 | PIN_OUTPUT		) /* D18 UART1_CTSN		7   GPIO0_12->DO_CPU_HLK1 */
 			0x17c ( MUX_MODE7 | PIN_OUTPUT		) /* D17 UART1_RTSN		7   GPIO0_13->DO_CPU_HLK2 */
 			0x1b4 ( MUX_MODE2 | PIN_INPUT		) /* D14 XDMA_EVENT_INTR1       2   tclkin->CNT-FAN->Counter_input */
@@ -234,6 +236,8 @@
 			0x060 ( MUX_MODE7 | PIN_INPUT_PULLUP	) /* V16 GPMC_A8		7   GPIO1_24->nINT_MOD->Card_interrupt_request */
 			0x064 ( MUX_MODE7 | PIN_INPUT		) /* U16 GPMC_A9		7   GPIO1_25->nINT_EXT*/
 			0x080 ( MUX_MODE7 | PIN_OUTPUT		) /* U9 GPMC_CSN1		7   GPIO1_30->nBOOT_END->Boot_process_end */
+			0x16c ( MUX_MODE7 | PIN_OUTPUT_PULLUP	) /* E17 UART0_RTSN		7   GPIO1_9->LED-K-BUS-G */
+			0x084 ( MUX_MODE7 | PIN_OUTPUT_PULLUP 	) /* V9 GPMC_CSN2		7   GPIO1_31->LED-K-BUS-R */
 		>;
 	};
 
@@ -246,20 +250,22 @@
 			0x0ac ( MUX_MODE7 | PIN_INPUT		) /* R4 LCD_DATA3		7   GPIO2_9->SYSBOOT3*/
 			0x0b0 ( MUX_MODE7 | PIN_INPUT		) /* T1 LCD_DATA4		7   GPIO2_10->SYSBOOT4*/
 			0x0b4 ( MUX_MODE7 | PIN_INPUT		) /* T2 LCD_DATA5		7   GPIO2_11->SYSBOOT5*/
-			0x0b8 ( MUX_MODE7 | PIN_INPUT		) /* T3 LCD_DATA6		7   GPIO2_12->SYSBOOT6*/
-			0x0bc ( MUX_MODE7 | PIN_INPUT		) /* T4 LCD_DATA7		7   GPIO2_13->SYSBOOT7*/
-			0x0c0 ( MUX_MODE7 | PIN_INPUT		) /* U1 LCD_DATA8		7   GPIO2_14->SYSBOOT8*/
-			0x0c4 ( MUX_MODE7 | PIN_INPUT		) /* U2 LCD_DATA9		7   GPIO2_15->SYSBOOT9*/
+			0x0b8 ( MUX_MODE7 | PIN_INPUT		) /* T3 LCD_DATA6		7   GPIO2_12->ERRST6*/
+			0x0bc ( MUX_MODE7 | PIN_INPUT		) /* T4 LCD_DATA7		7   GPIO2_13->ERRST7*/
+			0x0c0 ( MUX_MODE7 | PIN_INPUT		) /* U1 LCD_DATA8		7   GPIO2_14->ERRST8*/
+			0x0c4 ( MUX_MODE7 | PIN_INPUT		) /* U2 LCD_DATA9		7   GPIO2_15->ERRST9*/
 			0x0ec ( MUX_MODE7 | PIN_OUTPUT		) /* R6 LCD_AC_BIAS_EN		7   GPIO2_25->EN-V-FUSED->Switch_UI */
 			0x134 ( MUX_MODE7 | PIN_OUTPUT		) /* L17 MII1_RXD3		7   GPIO2_18->UO-HLK1->1._universal_HLK_output */
 			0x138 ( MUX_MODE7 | PIN_OUTPUT		) /* L16 MII1_RXD2		7   GPIO2_19->UO-HLK2->2._universal_HLK_output */
+			0x140 ( MUX_MODE7 | PIN_INPUT_PULLUP	) /* M16 GMII1_RXD0		7   GPIO2_21->KBUS-nIRQ */
 			//0x0e0 ( MUX_MODE7 | PIN_INPUT		  ) /* U5 LCD_VSYNC		  7   GPIO2_22->nINT-DALI->Interrupt_request_from_DALI */
 		>;
 	};
 
 	gpio3_pins: pinmux_gpio3_pins {
 		pinctrl-single,pins = <
-			0x118 ( MUX_MODE7 | PIN_OUTPUT		) /* J17 MII1_RX_DV		7   GPIO3_4->KBUS-nRST */
+			0x114 ( MUX_MODE7 | PIN_OUTPUT_PULLUP	) /* J16 GMII1_TXEN		7   GPIO3_3->KBUS-nSYNC */
+			0x118 ( MUX_MODE7 | PIN_OUTPUT_PULLUP 	) /* J17 GMII1_RXDV		7   GPIO3_4->KBUS-nRST */
 			0x12c ( MUX_MODE7 | PIN_OUTPUT		) /* K18 MII1_TX_CLK		7   GPIO3_9->DO-PI->digital_PI_output */
 			0x1a0 ( MUX_MODE7 | PIN_INPUT_PULLUP	) /* B12 MCASP0_ACLKR		7   GPIO3_18->MMC0_nCD->SD_card_detect */
 			0x1a4 ( MUX_MODE7 | PIN_INPUT_PULLDOWN	) /* C13 MCASP0_FSR		7   GPIO3_19->CUST_RES->Customer_reset */
@@ -285,6 +291,13 @@
 		>;
 	};
 
+	uart2_pins: pinmux_uart2_pins {
+		pinctrl-single,pins = <
+			0x10c ( MUX_MODE6 | PIN_INPUT		) /* H17 GMII1_CRS		6   UART2_RXD */
+			0x110 ( MUX_MODE6 | PIN_OUTPUT 		) /* J15 GMII1_RXER		6   UART2_TXD */
+		>;
+	};
+
 	uart3_pins: pinmux_uart3_pins {
 		pinctrl-single,pins = <
 			0x0c8 ( MUX_MODE6 | PIN_INPUT		) /* U3 LCD_DATA10		6   UART3_CTS->DALI_clear_to_send */
@@ -351,8 +364,8 @@
 			0x168 ( MUX_MODE4 | PIN_INPUT_PULLDOWN	) /* E18 UART0_CTSN		4   SPI1_D0->SPI1_MISO */
 			0x190 ( MUX_MODE3 | PIN_INPUT		) /* A13 MCASP0_ACLKX		3   SPI1_SCLK->Serial_clock */
 			0x198 ( MUX_MODE3 | PIN_OUTPUT		) /* D12 MCASP0_AXR0		3   SPI1_D1->SPI1_MOSI */
-			0x1b0 ( MUX_MODE4 | PIN_OUTPUT		) /* A15 XDMA_EVENT_INTR0	4   SPI1_CS1->SPI1_nCS1->Chip_select_fram */
-			//0x19c ( MUX_MODE4 | PIN_OUTPUT	  ) /* C12 MCASP0_AHCLKR	  4   SPI1_nCS0/spi1_cs0->Chip_select_kbus */
+			0x1b0 ( MUX_MODE4 | PIN_OUTPUT		) /* A15 XDMA_EVENT_INTR0	4   SPI1_nCS1->Chip_select_fram */
+			0x19c ( MUX_MODE3 | PIN_OUTPUT		) /* C12 MCASP0_AHCLKR		3   SPI1_nCS0->Chip_select_kbus */
 		>;
 	};
 
@@ -460,15 +473,14 @@
 
 	// FRAM - FM25CL64B-G
 	at25@0 {
-	compatible = "atmel,at25", "st,m95256";
-	reg = <0>;
-	spi-max-frequency = <20000000>;
-	spi-cpha;
-	spi-cpol;
-
-	pagesize = <8>;
-	size = <8192>;
-	address-width = <16>;
+		compatible = "atmel,at25", "st,m95256";
+		spi-max-frequency = <10000000>;
+		spi-cpha;
+		spi-cpol;
+		reg = <0>;
+		pagesize = <8>;
+		size = <8192>;
+		address-width = <16>;
 	};
 };
 
@@ -478,11 +490,18 @@
 	pinctrl-0 = <&spi1_pins>;
 
 	/* ATmega ISP */
-	spi1_0: spidev@0 {
+	spi2_0: spidev@0 {
 		compatible = "wago,wds";
 		reg = <0>;
 		spi-max-frequency = <10000000>; // FIXME
 	};
+
+	// KBUS Spidev
+	spi2_1: spidev@0 {
+		compatible = "wago,srckbus";
+		reg = <1>;
+		spi-max-frequency = <10000000>;
+	};
 };
 
 &i2c0 {
@@ -684,6 +703,13 @@
 	linux,rs485-enabled-at-boot-time;
 };
 
+&uart2 {
+	/* KBUS */
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2_pins>;
+};
+
 &uart3 {
 	/* DALI */
 	status = "okay";
diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 16b3581..4aa2fa6 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -697,6 +697,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
 	{ .compatible = "wago,wds" },
+	{ .compatible = "wago,srckbus" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

